beetsplug/vocadb/__init__.py

```diff
@@ -0,0 +1,1218 @@
+                    remote_lyrics.translation_type
+                )
+                value: str | None = remote_lyrics.value
+                # get the intersection
+                culture_codes: set[str] | None = remote_lyrics.culture_codes
+                if culture_codes:
+                    culture_codes &= {
+                        "en",
+                        "ja",
+                    }
+
+                if not culture_codes:
+                    if (
+                        not translated_lyrics
+                        and language_preference
+                        == ContentLanguagePreference.ROMAJI
+                        and remote_translation_type == TranslationType.ROMANIZED
+                    ):
+                        lyrics = value
+                    continue
+
+                if "en" in culture_codes:
+                    if remote_translation_type == TranslationType.ORIGINAL:
+                        script = "Latn"
+                        language = "eng"
+                    if (
+                        translated_lyrics
+                        or language_preference
+                        == ContentLanguagePreference.ENGLISH
+                    ):
+                        lyrics = value
+                    continue
+
+                if "ja" in culture_codes:
+                    if remote_translation_type == TranslationType.ORIGINAL:
+                        script = "Jpan"
+                        language = "jpn"
+                    if (
+                        not translated_lyrics
+                        and language_preference
+                        == ContentLanguagePreference.JAPANESE
+                    ):
+                        lyrics = value
+
+            if not lyrics and remote_lyrics_list:
+                lyrics = self.get_fallback_lyrics(remote_lyrics_list)
+
+        return script, language, lyrics
+
+    def get_fallback_lyrics(
+        self,
+        remote_lyrics_list: list[LyricsForSongContract],
+    ) -> str | None:
+        language_preference: ContentLanguagePreference = (
+            self.instance_config.language
+        )
+        remote_lyrics: LyricsForSongContract
+        if language_preference == ContentLanguagePreference.ENGLISH:
+            for remote_lyrics in remote_lyrics_list:
+                culture_codes: set[str] | None
+                if (
+                    culture_codes := remote_lyrics.culture_codes
+                ) and "en" in culture_codes:
+                    return remote_lyrics.value
+            language_preference = ContentLanguagePreference.ROMAJI
+        if language_preference == ContentLanguagePreference.ROMAJI:
+            for remote_lyrics in remote_lyrics_list:
+                if remote_lyrics.translation_type == TranslationType.ROMANIZED:
+                    return remote_lyrics.value
+        return remote_lyrics_list[0].value if remote_lyrics_list else None
```

Comment:
Do we know if the first item always has translation_type == TranslationType.ORIGINAL?

Reply:
From what I know, no. While unlikely, a track can have translated lyrics while missing original lyrics.

Reply:
After looking a bit at how VocaDB does lyrics, doesn't seem like they're sorted to always be index 0. They're just sorted in the order they are added.

Reply:
so there should probably be another condition, right?

```suggestion
        if language_preference == ContentLanguagePreference.DEFAULT
            for remote_lyrics in remote_lyrics_list:
                if remote_lyrics.translation_type == TranslationType.ORIGINAL:
                    return remote_lyrics.value
        return remote_lyrics_list[0].value if remote_lyrics_list else None
```

Reply:
Yea that would work.
Though you could even drop the last line and just `return None`, since by that point we have tested all of the languages the plugin officially supports.